server/src/main/java/org/apache/cassandra/sidecar/acl/authentication/MutualTlsAuthenticationHandler.java

```diff
@@ -88,10 +90,18 @@ public void authenticate(RoutingContext ctx, Handler<AsyncResult<User>> handler)
 
                         List<String> identities = extractIdentities(authN.result());
                         List<String> roles = extractCassandraRoles(identities);
-                        if (!roles.isEmpty())
+                        String roleIntended = ctx.request().getHeader(AUTH_ROLE);
+
+                        if (roleIntended != null && !roleIntended.isEmpty() && !roles.contains(roleIntended))
                         {
-                            authN.result().attributes().put(CASSANDRA_ROLES_ATTRIBUTE_NAME, roles);
+                            String errMsg = String.format("User not authorized for role %s", roleIntended);
```

Comment:
I think we should give more information here. Both pieces of information are provided by the user, identities + role
```suggestion
                            String errMsg = String.format("None of the identities %s are authorized for role %s", identities, roleIntended);
```